tracer/src/Datadog.Trace/Span.ISpan.cs

```diff
@@ -0,0 +1,117 @@
+// <copyright file="Span.ISpan.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// A Span represents a logical unit of work in the system. It may be
+    /// related to other spans by parent/children relationships. The span
+    /// tracks the duration of an operation as well as associated metadata in
+    /// the form of a resource name, a service name, and user defined tags.
+    /// </summary>
+    public partial class Span : ISpan
+    {
+        /// <summary>
+        /// Gets or sets operation name
+        /// </summary>
+        string ISpan.OperationName
+        {
+            get => OperationName;
+            set => OperationName = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the resource name
+        /// </summary>
+        string ISpan.ResourceName
+        {
+            get => ResourceName;
+            set => ResourceName = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the type of request this span represents (ex: web, db).
+        /// Not to be confused with span kind.
+        /// </summary>
+        /// <seealso cref="SpanTypes"/>
+        /// <summary>
+        /// Gets or sets the resource name
+        /// </summary>
+        string ISpan.Type
+        {
+            get => Type;
+            set => Type = value;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this span represents an error
+        /// </summary>
+        bool ISpan.Error
+        {
+            get => Error;
+            set => Error = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the service name.
+        /// </summary>
+        string ISpan.ServiceName
+        {
+            get => ServiceName;
+            set => ServiceName = value;
+        }
```

Comment:
nit: `<inheritdoc />`
```suggestion
        /// <inheritdoc />
        string ISpan.OperationName
        {
            get => OperationName;
            set => OperationName = value;
        }

        /// <inheritdoc />
        string ISpan.ResourceName
        {
            get => ResourceName;
            set => ResourceName = value;
        }

        /// <inheritdoc />
        string ISpan.Type
        {
            get => Type;
            set => Type = value;
        }

        /// <inheritdoc />
        bool ISpan.Error
        {
            get => Error;
            set => Error = value;
        }

        /// <inheritdoc />
        string ISpan.ServiceName
        {
            get => ServiceName;
            set => ServiceName = value;
        }
```